docs/components/concepts/messages.md

```diff
@@ -49,6 +49,13 @@ zbctl publish message "Money collected" --correlationKey "order-123" --ttl 1h
    </p>
  </details>
 
+## Message expiration
+
+When a buffered message could not be correlated to a process instance in the
+given time-to-live, it gets removed from the buffer afterwards. Internally, the
+MessageObserver runs once a minute and exports an event, that the message
+expired.
```

Comment:
❓ Do we want to document this in concepts? 

IMO, this is an implementation detail. Perhaps we should change the description of the TTL in the `Message Buffering` section to 
> "A message has a **time-to-live** (**TTL**) which specifies the minimum duration that it's buffered".

We don't and cannot guarantee how soon after the TTL expires, the message is discarded. The MessageTimeToLiveChecker's interval is just an implementation detail currently in place. In my opinion, it is not part of our API, so we should not discuss it in the documentation, as we may change this later.

Note that I'm currently working on allowing users to configure the MessageTimeToLiveChecker's interval. We can describe that users can adjust this in the config file. There it should be clear what the effect of that adjustment is.

❌ I'm also not convinced of event exporting in the Concepts section, as it is a Self-Managed-only feature.

Reply:
Hi @korthout, from my developer perspective it is important to know, that at some point in time I can get information about the expired message. 

I discussed it with a customer who wants to react on this expired message (send it again or announce the expiration to somebody else). Eventually, the message expiration event should be available via Operate API.

At the moment, I don't care about the minute. Could be longer or shorter or configurable.

For me, the important learning of the underlying issue was, that the event happens after some delay, and it is worth to mention the delay and the source where it is coming from.

Even if I can't change the minute by myself right now, it may be useful to look up the implementation and estimate the effort, that a feature request to make in configurable would take. With this argument, the section gives me all information that I was looking for.

Reply:
Sorry @ingorichtsmeier, I missed your reply.

>it is important to know, that at some point in time I can get information about the expired message

I find it difficult to document it like this. This functionality only exists in Self-Managed (exporters are not available to SaaS users), so documenting it as a feature in the main Concepts would be confusing SaaS users. If we would document it there, it would require a special Self-Managed-only notice.

>Eventually, the message expiration event should be available via Operate API.

That would be a great addition 👍 

>a feature request to make in configurable 

👍 I'm currently implementing this, it is part of the [message expiration performance improvements](https://github.com/camunda/zeebe/issues/11591). Since that is considered a bug, we will backport these changes to all supported minor versions.

>With this argument, the section gives me all information that I was looking for.

❓ Do I understand correctly that you're happy with the configuration approach as alternative documentation?